Description: only emit warnings about overriding LSB default with symlink
 state for the scripts in insserv arg list.
From: Kel Modderman <kel@otaku42.de>
---
--- a/insserv.c
+++ b/insserv.c
@@ -3100,8 +3100,9 @@ int main (int argc, char *argv[])
 			     * of the current script.
 			     */
 			    if (!defaults && service->start->lvl != 0) {
-				warn("warning: current start runlevel(s) (%s) of script `%s' overwrites defaults (empty).\n",
-				     lvl2str(service->start->lvl), d->d_name);
+				if (!del && isarg && !(argr[curr_argc]))
+				    warn("warning: current start runlevel(s) (%s) of script `%s' overwrites defaults (empty).\n",
+					 lvl2str(service->start->lvl), d->d_name);
 				script_inf.default_start = lvl2str(service->start->lvl);
 			    }
 			}
@@ -3162,8 +3163,9 @@ int main (int argc, char *argv[])
 			     * of the current script.
 			     */
 			    if (!defaults && service->stopp->lvl != 0) {
-				warn("warning: current stop runlevel(s) (%s) of script `%s' overwrites defaults (empty).\n",
-				     lvl2str(service->stopp->lvl), d->d_name);
+				if (!del && isarg && !(argr[curr_argc]))
+				    warn("warning: current stop runlevel(s) (%s) of script `%s' overwrites defaults (empty).\n",
+					 lvl2str(service->stopp->lvl), d->d_name);
 				script_inf.default_stop = lvl2str(service->stopp->lvl);
 			    }
 			}
